packages/big-design/src/components/Worksheet/Worksheet.tsx

```diff
@@ -17,21 +17,35 @@ const InternalWorksheet = <T extends WorksheetItem>({
   onChange,
   onErrors,
 }: WorksheetProps<T>): React.ReactElement<WorksheetProps<T>> => {
-  const rows = useStore(useMemo(() => (state) => state.rows, []));
+  const tableRef = createRef<HTMLTableElement>();
+
   const setRows = useStore((state) => state.setRows);
+  const setColumns = useStore((state) => state.setColumns);
+  const setTableRef = useStore((state) => state.setTableRef);
 
+  const rows = useStore(useMemo(() => (state) => state.rows, []));
   const editedCells = useStore(useMemo(() => (state) => state.editedCells, []));
   const invalidCells = useStore(useMemo(() => (state) => state.invalidCells, []));
 
+  const { handleKeyDown } = useKeyEvents();
+
   // Create a new reference since state mutates rows to prevent unecessary rerendering
   useEffect(() => setRows([...items]), [items, setRows]);
 
+  // Save columns to use in navigation
+  useEffect(() => setColumns(columns), [columns, setColumns]);
+
+  // Save ref to focus table when necessary
+  useEffect(() => setTableRef(tableRef.current), [setTableRef, tableRef]);
+
+  // Call onChange when a cell is edited
```

Comment:
I'm not sure if this comment (and below) are adding any meaningful information for the developers working in this file